broker/src/main/java/io/zeebe/broker/transport/backpressure/PartitionAwareRequestLimiter.java

```diff
@@ -42,7 +43,12 @@ private PartitionAwareRequestLimiter() {
   }
 
   private PartitionAwareRequestLimiter(final Supplier<Limit> limitSupplier) {
-    limiterSupplier = i -> CommandRateLimiter.builder().limit(limitSupplier.get()).build(i);
+    limiterSupplier =
+        i ->
+            CommandRateLimiter.builder()
+                .clock(new IncreasingClock())
```

Comment:
I don't see any benefit in making this configurable. If the user's clock doesn't cause this bug, this won't change anything, the measurements will be exactly the same. Besides, how would the user know to enable this? A user can't know that 'System.nanoTime()` will cause us problems a priori, only after a crash.

Reply:
There is a slight overhead in using IncreasingClock because of additional synchronized access. The user would know to enable this, if the other log logs it as a warning. The system wouldn't crash anymore, right? Because now you catch the exception and logs it. My expectation is that users test it on production hardware before actually running it on production.